### PR TITLE
Fixed bug in the calculation of the header block fragment length

### PR DESCRIPTION
**Pull request by bryancall**
Co-authored-by: Masaori Koshiba <masaori@apache.org>

### CODE REVIEW

**Comment by bryancall** [comment]
[approve ci clang-analyzer]

**Comment by bryancall** [comment]
[approve ci autest]

**Comment by bryancall** [comment]
[approve ci rat]

**Comment by bryancall** [comment]
[approve ci clang-format]

**Comment by bryancall** [comment]
[approve ci autest]

**Comment by bryancall** [comment]
[approve ci clang-analyzer]

**Comment by bryancall** [comment]
[approve ci rat]

**Comment by bryancall** [comment]
[approve ci clang-format]

File: proxy/http2/Http2ConnectionState.cc [hunk 1]
```diff
@@ -328,11 +328,19 @@ rcv_headers_frame(Http2ConnectionState &cstate, const Http2Frame &frame)
     }
   }
 
+  stream->header_blocks_length = header_block_fragment_length;
+
+  // ATS advertises SETTINGS_MAX_HEADER_LIST_SIZE as a limit of total header blocks length. (Details in [RFC 7560] 10.5.1.)
+  // Make it double to relax the limit in cases of 1) HPACK is used naively, or 2) Huffman Encoding generates large header blocks.
+  // The total "decoded" header length is strictly checked by hpack_decode_header_block().
+  if (stream->header_blocks_length > std::max(Http2::max_header_list_size, Http2::max_header_list_size * 2)) {
```
Comment (zizhong) [inline proxy/http2/Http2ConnectionState.cc:336]: This is confusing.
What do we need max(A, 2 * A)?
Possible reasons could be,
1. Wrong variables
2. A < 0.

Comment (bryancall) [reply proxy/http2/Http2ConnectionState.cc:336]: I am doing that incase we have an overflow.

Comment (zizhong) [reply proxy/http2/Http2ConnectionState.cc:336]: TIL!

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**